src/tag/constraints.test.ts

```diff
@@ -0,0 +1,49 @@
+import { describe, expect, it } from 'vitest';
+
+import { TAG_NAME_PATTERN } from './constraints.js';
+
+describe('TAG_NAME_PATTERN', () => {
+  it('alphanumerical', () => {
+    expect(new RegExp(TAG_NAME_PATTERN).test('mytag')).toBeTruthy();
+    expect(new RegExp(TAG_NAME_PATTERN).test('My tag')).toBeTruthy();
+    expect(new RegExp(TAG_NAME_PATTERN).test('MYTAG')).toBeTruthy();
+    expect(new RegExp(TAG_NAME_PATTERN).test('My wonderful tag')).toBeTruthy();
+    expect(new RegExp(TAG_NAME_PATTERN).test('tag-with-âccent.')).toBeTruthy();
+
+    expect(new RegExp(TAG_NAME_PATTERN).test('2034')).toBeTruthy();
+    expect(new RegExp(TAG_NAME_PATTERN).test('my-tag-302')).toBeTruthy();
+    expect(new RegExp(TAG_NAME_PATTERN).test('a')).toBeTruthy();
+    expect(new RegExp(TAG_NAME_PATTERN).test('1')).toBeTruthy();
+  });
```

Comment:
You can refactor this test using the `it.each()` from vitest. Provide an array as the input and the function will be executed for each value in the array.